config/samples/pipeline_v1beta1_pipeline.yaml

```diff
@@ -5,48 +5,56 @@ metadata:
-        - name: 'buildPush'
-          taskRef:
-              name: build-push-task
-          sourceBindings:
-              - inputName: 'workspace'
-                sourceKey: 'wizzbang'
-          artifactStoreBindings:
-              - storeName: 'registry'
-                storeKey: 'stagingRegistry'
-          params:
-              - name: 'pathToDockerfile'
-                value: 'build/Dockerfile'
-        - name: 'integrationTest'
-          taskRef:
-              name: integration-test-wizzbang-task
-          sourceBindings:
-              - inputName: 'repoUnderTest'
-                sourceKey: 'wizzbang'
-          paramBindings:
-              - inputName: testImage
-                taskName: buildPush
-                taskOutputName: builtImage
-        - name: 'deploy'
-          taskRef:
-              name: deploy-with-helm
-          params:
-              - name: pathToHelmCharts
-                value: deploy/helm
-          sourceBindings:
-              - inputName: 'repoToDeploy'
-                sourceKey: 'wizzbang'
-          clusterBindings:
-              - clusterName: prod
-          paramBindings:
-              - inputName: image
-                taskName: buildPush
-                taskOutputName: builtImage
-          prevTasks: ['integrationTest']
+      - name: 'test'
+        taskRef:
+          name: test-wizzbang-task
+        inputsourceBindings:
+          - name: repoUnderTest
+            sourceKey: wizzbang
+      - name: 'buildPush'
+        taskRef:
+          name: build-push-task
+        inputSourceBindings:
+          - name: 'workspace'
+            sourceKey: 'wizzbang'
+            passedConstraint: [test-wizzbang-task]
+        outputSourceBindings:
+          - name: registry
+            sourceKey: wizzbangStagingImage
+        params:
+          - name: pathToDockerfile
+            value: build/Dockerfile
+      - name: integrationTest
+        taskRef:
+          name: integration-test-wizzbang-task
+        inputSourceBindings:
+          - name: repoUnderTest
+            sourceKey: wizzbang
+            passedConstraint: [test]
+        outputSourceBindings:
+          - name: registry
+            sourceKey: wizzbangStagingImage
+            passedConstraint: [build-push-task]
+      - name: deploy
+        taskRef:
+            name: deploy-with-helm
+        params:
+          - name: pathToHelmCharts
+            value: deploy/helm
+        inputSourceBindings:
+          - name: repoToDeploy
+            sourceKey: wizzbang
+            passedConstraint: [integration-test-wizzbang-task]
+        clusterBindings:
+          - clusterName: prod
+        outputSourceBindings:
+          - name: registry
+            sourceKey: wizzbangStagingImage
+            passedConstraint: [build-push-task]
+    resources:
+    - name: wizzbang
+      resourceRef:
+        name: wizzbang
+    - name: wizzbangStagingImage
+      resourceRef:
+        name: wizzbangStagingImage   
```

Comment:
i think that this reference belongs in the `PipelineRun`, since that would be the place where you'd bring everything together (otherwise everytime you use this Pipeline, you have to use the same resources). but we can iterate on this!